src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -107,6 +107,11 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
         const client = createAbstractPostgresClient(serverType, treeItem.root);
         const result: FirewallRuleListResult = (await client.firewallRules.listByServer(nonNullProp(treeItem, 'resourceGroup'), nonNullProp(treeItem, 'azureName')))._response.parsedBody;
         const publicIp: string = await getPublicIp();
-        return (result.some(async value => value.startIpAddress === publicIp));
```

Comment:
This check is not working correctly. This was working in some cases where the result is empty and it is resolving directly to false. `async value => value.startIpAddress === publicIp` is not resolving to checking each array object.